vmdb/app/controllers/application_controller.rb

```diff
@@ -438,18 +438,8 @@ def saved_report_paging
   end
 
   def build_vm_host_array
-    @temp[:tree_hosts] = []
-    @temp[:tree_vms]   = []
-    if !@sb[:tree_hosts].blank?
-      @sb[:tree_hosts].each do |h|
-        @temp[:tree_hosts].push(Host.find_by_id(h))
-      end
-    end
-    if !@sb[:tree_vms].blank?
-      @sb[:tree_vms].each do |v|
-        @temp[:tree_vms].push(Vm.find_by_id(v))
-      end
-    end
+    @temp[:tree_hosts] = (@sb[:tree_hosts_hash] || {}).collect { |k,_| Host.find_by_id(k) }
+    @temp[:tree_vms]   = (@sb[:tree_vms_hash]   || {}).collect { |k,_| Vm.find_by_id(k) }
```

Comment:
We can do that --> will do that, it's a cheap change. 

But the bottleneck is not here :-(

After this PR, the slowest part seems to be the rendering of the of the quadicons. It's several hundred partials that each in turn call a partial. It takes ages. I would like to try to rewrite it to helper functions. But that's out of scope for this BZ.